Eventing Testkit: expectN not unwrapping json

When trying to assert on messages retrieved with expectN:

E.g.

```java
    List<EventingTestKit.Message<?>> remainingEvents = topic.expectN(2);
    var bid1Event = remainingEvents.get(0).expectType(BidAccepted.class);
```

This error appears:
`java.lang.AssertionError: Expected class com.example.domain.AuctionEntity$AuctionEvent$BidAccepted, found class java.lang.String ({"bidId":"71255c25-04c8-476e-8948-3297236ff8ac","value":30})`

--- a/sdk/java-sdk-protobuf-testkit/src/main/scala/kalix/javasdk/testkit/impl/EventingTestKitImpl.scala
+++ b/sdk/java-sdk-protobuf-testkit/src/main/scala/kalix/javasdk/testkit/impl/EventingTestKitImpl.scala
@@ -266,8 +266,9 @@ private[testkit] class TopicImpl(
     val scalaPb = ScalaPbAny(typeUrlFor(metadata), msg.getMessage.payload)
 
     val decodedMsg = if (typeUrlFor(metadata).startsWith(JsonSupport.KALIX_JSON)) {
-      val objectMapper = JsonSupport.getObjectMapper
-      objectMapper.readerFor(clazz).readValue(msg.getMessage.payload.toByteArray)
+      JsonSupport.getObjectMapper
+        .readerFor(clazz)
+        .readValue(msg.getMessage.payload.toByteArray)
     } else {
       codec.decodeMessage(scalaPb)
     }
@@ -397,6 +398,7 @@ private[testkit] object TestKitMessageImpl {
     val bt = BoxedType(clazz)
     payload match {
       case m if bt.isInstance(m) => m.asInstanceOf[T]
+      case m: String             => JsonSupport.getObjectMapper.readerFor(clazz).readValue(m)
       case m                     => throw new AssertionError(s"Expected $clazz, found ${m.getClass} ($m)")
     }
   }

